cli/pkg/utils/hash.go

```diff
@@ -35,7 +35,7 @@ func GenerateHash(inputType utils.SourceType, source string) (string, error) {
 		return "", fmt.Errorf("failed to get absolute path of the source %s: %v", source, err)
 	}
 	switch inputType {
-	case utils.IMAGE:
+	case utils.IMAGE, utils.DOCKERARCHIVE, utils.OCIARCHIVE:
```

Comment:
I think that in the case of archive we need to calcualte the hash of the archive, similar to what we are doing in `dir`, wdyt?
Otherwise, what will be the hash in this case? It is not the hash of the image, right?

Reply:
docker-archive and oci-archive both use the manifest digest is the hash. This are output by trivy and by syft so no need to hash them manually here.

Reply:
From what I can see, when running trivy on docker-archive, the sbom output is similar to the local image scanning, no manifest digest.
```
$ trivy image --input ~/openssh-server.tar -f cyclonedx
{
  "$schema": "http://cyclonedx.org/schema/bom-1.5.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.5",
  "serialNumber": "urn:uuid:e4b2c76c-793f-47d7-9e78-bb8c021ddb46",
  "version": 1,
  "metadata": {
    "timestamp": "2023-09-13T12:59:26+00:00",
    "tools": [
      {
        "vendor": "aquasecurity",
        "name": "trivy",
        "version": "0.44.1"
      }
    ],
    "component": {
      "bom-ref": "7c89e91a-c076-4a5c-a5d5-48c07621e0c7",
      "type": "container",
      "name": "/Users/idanf/openssh-server.tar",
      "properties": [
        {
          "name": "aquasecurity:trivy:DiffID",
          "value": "sha256:d93a6ceac0ee120f96934ca789fadea962782781c889c1e96e740932320db5cd,sha256:953f06f22e4159de6072836ed4357329d8bc0166b2ec8210cd7f8444c5b74460,sha256:19ce83e2a83aad8244353297151f1b973e80e784f4839ce8e4336b718901ed68,sha256:5a292a7233024a61a86275fcfca78f61562bd437e81da25e68035b18b1608334,sha256:9e4d983f42c8686a0ad4b004d98d4e4f08c21ff27d6c83c91c3325194f934d50,sha256:22338be56fb3d75918cf181a10c276b8227c2650eac776d1a3cfbb604099e0b2,sha256:fbe1e86efa0922cb5232bdd33b99cce01417555504f4242f8d5d8084cb01db04"
        },
        {
          "name": "aquasecurity:trivy:ImageID",
          "value": "sha256:ec526013bc0381a656f4fd639e39913e1dc0d75fecd04a9c1e1ffe9cb0a67469"
        },
        {
          "name": "aquasecurity:trivy:SchemaVersion",
          "value": "2"
        }
      ]
    }
  },
```

Reply:
Decided to handle this and local images all together in a separate ticket. It seems we'll need a common solution as trivy and syft do not agree on the sha for an image.